src/main/java/org/broadinstitute/consent/http/resources/UserResource.java

```diff
@@ -258,30 +282,69 @@ public Response addRoleToUser(@Auth AuthUser authUser, @PathParam("userId") Inte
         }
     }
 
+    private static boolean signingOfficialMeetsRequirements(Integer roleId, User activeUser, User user) {
+        return activeUser.hasUserRole(UserRoles.SIGNINGOFFICIAL)
+                && Objects.nonNull(activeUser.getInstitutionId())
+                && UserRoles.isValidSoAdjustableRoleId(roleId)
+                && (Objects.equals(user.getInstitutionId(), activeUser.getInstitutionId()) ||
+                Optional.ofNullable(user.getInstitutionId()).isEmpty());
+    }
+
+    private Response getUserResponse(AuthUser authUser, Integer userId) {
+        JsonObject userJson = userService.findUserWithPropertiesByIdAsJsonObject(authUser, userId);
+        return Response.ok().entity(gson.toJson(userJson)).build();
+    }
+
     @DELETE
     @Path("/{userId}/{roleId}")
     @Produces("application/json")
-    @RolesAllowed({ADMIN})
+    @RolesAllowed({ADMIN, SIGNINGOFFICIAL})
     public Response deleteRoleFromUser(@Auth AuthUser authUser, @PathParam("userId") Integer userId, @PathParam("roleId") Integer roleId) {
+        UserRoles targetRole = UserRoles.getUserRoleFromId(roleId);
+        if (Objects.isNull(targetRole)) {
+            return Response.status(HttpStatusCodes.STATUS_CODE_BAD_REQUEST).build();
+        }
         try {
+            User activeUser = userService.findUserByEmail(authUser.getEmail());
             User user = userService.findUserById(userId);
-            if (!UserRoles.isValidNonDACRoleId(roleId)) {
-                throw new BadRequestException("Invalid Role Id");
-            }
-            List<Integer> currentUserRoleIds = user.getUserRoleIdsFromUser();
-            if (!currentUserRoleIds.contains(roleId)) {
-                JsonObject userJson = userService.findUserWithPropertiesByIdAsJsonObject(authUser, userId);
-                return Response.ok().entity(gson.toJson(userJson)).build();
+            if (activeUser.hasUserRole(UserRoles.ADMIN)) {
+                if (!UserRoles.isValidNonDACRoleId(roleId)) {
+                    throw new BadRequestException("Invalid Role Id");
+                }
+                return doDelete(authUser, userId, roleId, activeUser, user);
+            } else if (activeUser.hasUserRole(UserRoles.SIGNINGOFFICIAL)) {
+                if (!UserRoles.isValidSoAdjustableRoleId(roleId)) {
+                    throw new ForbiddenException("A Signing Official may only remove the following role ids: [6, 7, 8] ");
```

Comment:
We should document this failure case in `api-docs.yaml` ... look near line 1916. One more thing we're doing is trying to slim down this file by moving the method docs into separate yamls in `src/main/resources/assets/paths`. The `User` schema is already defined in `src/main/resources/assets/schemas` so there shouldn't be any additional code for that.